lib/charms/traefik_k8s/v1/ingress.py

```diff
@@ -221,51 +252,51 @@ def __init__(self, charm: CharmBase, relation_name: str = DEFAULT_RELATION_NAME)
                 (defaults to "ingress").
         """
         super().__init__(charm, relation_name)
-        self.framework.observe(
-            self.charm.on[relation_name].relation_joined, self._share_version_info
-        )
-
-    def _share_version_info(self, event):
-        """Backwards-compatibility shim for version negotiation.
-
-        Allows older versions of IPA (requirer side) to interact with this
-        provider without breaking.
-        Will be removed in a future version of this library.
-        Do not use.
-        """
-        relation = event.relation
-        if self.charm.unit.is_leader():
-            log.info("shared supported_versions shim information")
-            relation.data[self.charm.app]["_supported_versions"] = "- v1"
 
     def _handle_relation(self, event):
         # created, joined or changed: if remote side has sent the required data:
         # notify listeners.
         if self.is_ready(event.relation):
-            self.on.data_provided.emit(event.relation)
+            data = self._get_requirer_data(event.relation)
+            self.on.data_provided.emit(
+                event.relation,
+                data["name"],
+                data["model"],
+                data["port"],
+                data["host"],
+            )
 
     def _handle_relation_broken(self, event):
         self.on.data_removed.emit(event.relation)
 
     def wipe_ingress_data(self, relation: Relation):
         """Clear ingress data from relation."""
-        del relation.data[self.app]["data"]
+        del relation.data[self.app]["ingress"]
```

Comment:
`del` seems like a dirty solution. Wouldn't setting it to `None` do just as well?

Reply:
`del` is the official way to do it actually, setting it to `None` would hopefully raise an exception as the values are supposed to be strings. `del` is in fact an alias for `data[key] = ''`

Reply:
Ok, makes sense. Thanks